pylops/basicoperators/Spread.py

```diff
@@ -50,22 +50,61 @@ class Spread(LinearOperator):
+
     dtable : :obj:`np.ndarray`, optional
         Look-up table of decimals remainders for linear interpolation of size
         :math:`[n_{x_0} \times n_{t_0} \times n_x]` (if ``None`` use function
-        handle ``fh``)
-    fh : :obj:`np.ndarray`, optional
-        Function handle that returns a list of time indices (and fractional values
-        for spreading with linear interpolation) to be used for spreading/stacking
-        (if ``None`` use look-up table ``table``)
+        handle ``fh``). When provided, the ``data`` will be created as follows
+
+        .. code-block:: python
+
+            data[ix, table[ix0, it0, ix]]     += (1 - dtable[ix0, it0, ix]) * model[ix0, it0]
+            data[ix, table[ix0, it0, ix] + 1] +=      dtable[ix0, it0, ix]  * model[ix0, it0]
+
+        .. note::
+            When using ``table`` and ``dtable``, the elements of ``table`` indices must be
+            between 0 and :math:`n_{t_0} - 2`  (or ``numpy.nan``).
+
+    fh : :obj:`callable`, optional
+        If ``None`` will use look-up table ``table``. When provided, should be a
+        function which takes indices ``ix0`` and ``it0`` and returns
+        an array of size :math:`n_x` containing each respective time index.
+        Alternatively, if linear interpolation is required, it should output in
+        addition to the time indices, a weight for interpolation with linear
+        interpolation, to be used as follows
+
+        .. code-block:: python
+
+            data[ix, index]     += (1 - dindices[ix]) * model[ix0, it0]
+            data[ix, index + 1] +=      dindices[ix]  * model[ix0, it0]
+
+        where ``index`` refers to a time index in the first array returned by ``fh``
+        and ``dindices`` refers to the weight in the second array returned by ``fh``.
+
+        .. note::
+            When using ``fh`` with one output (time indices), the time indices must be
+            between 0 and :math:`n_{t_0} - 1` (or ``numpy.nan``). When using ``fh`` with two outputs
+            (time indices and weights), they must be within the between 0 and
+            :math:`n_{t_0} - 2` (or ``numpy.nan``).
+
     interp : :obj:`bool`, optional
-        Apply linear interpolation (``True``) or nearest interpolation
-        (``False``) during stacking/spreading along parametric curve. To be
-        used only if ``engine='numba'``, inferred directly from the number of
-        outputs of ``fh`` for ``engine='numpy'``
+        Use only if engine ``engine='numba'``. Apply linear interpolation (``True``)
+        or nearest interpolation (``False``) during stacking/spreading along
+        parametric curve.
+        When using ``engine="numpy"``, it will be inferred directly from ``fh`` or
+        the presence of ``dtable``.
     engine : :obj:`str`, optional
         Engine used for fft computation (``numpy`` or ``numba``). Note that
         ``numba`` can only be used when providing a look-up table
```

Comment:
Is this actually right? Numba can only be used with table? Also, what is the fft referring to?

Reply:
Oh fft is a mistake, also the other it used to be like this, not anymore